Mathlib/Algebra/Algebra/Subalgebra/Basic.lean

```diff
@@ -695,6 +695,16 @@ theorem inf_toSubsemiring (S T : Subalgebra R A) :
     (S ⊓ T).toSubsemiring = S.toSubsemiring ⊓ T.toSubsemiring :=
   rfl
 
+@[simp]
+theorem sup_toSubsemiring (S T : Subalgebra R A) :
+    (S ⊔ T).toSubsemiring = S.toSubsemiring ⊔ T.toSubsemiring := by
+  rw [← S.toSubsemiring.closure_eq, ← T.toSubsemiring.closure_eq, ← Subsemiring.closure_union]
+  change Subsemiring.closure (Set.range (algebraMap R A) ∪ (S ∪ T)) = Subsemiring.closure (S ∪ T)
```

Comment:
I don't think it's a great idea to pull this out of thin air. I would add
```
@[simp]
theorem adjoin_toSubsemiring (s : Set A) :
    (adjoin R s).toSubsemiring = Subsemiring.closure (Set.range (algebraMap R A) ∪ s) :=
  rfl
```
after `def adjoin (s : Set A) : Subalgebra R A := ...` and
```
theorem sup_def (S T : Subalgebra R A) : S ⊔ T = adjoin R (S ⊔ T : Set A) := rfl
```
somewhere after `instance : CompleteLattice (Subalgebra R A) where`, and then this can become
```suggestion
  simp_rw [sup_def, adjoin_toSubsemiring, Subalgebra.coe_toSubsemiring, Set.sup_eq_union]
```